### PR TITLE
Update to Ubuntu-mate-desktop

### PR DESCRIPTION
**Pull request by leestott**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by leestott** [comment]
@bmoore-msft this template fails but the main changes were making the template deployable at present the deployment link is incorrect.

**Comment by leestott** [comment]
@bmoore-msft this template now passes checks

**Comment by leestott** [comment]
@bmoore-msft  can this please be merged Template now validates and passes CI for Both Public and Gov

File: 101-ubuntu-mate-desktop-vscode/azuredeploy.json [hunk 1]
```diff
@@ -262,8 +245,8 @@
         "autoUpgradeMinorVersion": true,
         "settings": {
           "fileUris": [
-            "[concat(parameters('_artifactsLocation'), '/', variables('ScriptFileName'), parameters('_artifactsLocationSasToken'))]"
-          ]
+            "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/101-ubuntu-mate-desktop-vscode/infra.sh"
```
Comment (bmoore-msft) [inline 101-ubuntu-mate-desktop-vscode/azuredeploy.json]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts) for how to handle artifacts

Comment (leestott) [reply 101-ubuntu-mate-desktop-vscode/azuredeploy.json]: @bmoore-msft  thanks changes made and new PR

**Comment by leestott** [comment]
@bmoore-msft this is now much neater deployment

**Pull request merged**